src/metatrain/utils/scaler.py

```diff
@@ -0,0 +1,237 @@
+                        target_info_dict = {target_key: self.new_targets[target_key]}
+                        targets = remove_additive(
+                            systems,
+                            targets,
+                            additive_model,
+                            target_info_dict,
+                        )
+
+                    target_info = self.new_targets[target_key]
+                    if (
+                        target_info.quantity == "energy"
+                        and "positions" in target_info.gradients
+                    ):
+                        # special case: here we want to scale with respect to the forces
+                        # rather than the energies
+                        sum_of_squared_targets += torch.sum(
+                            targets[target_key].block().gradient("positions").values
+                            ** 2
+                        ).item()
+                        total_num_elements += (
+                            targets[target_key]
+                            .block()
+                            .gradient("positions")
+                            .values.numel()
+                        )
+                    else:
+                        sum_of_squared_targets += sum(
+                            torch.sum(block.values**2).item()
+                            for block in targets[target_key].blocks()
+                        )
+                        total_num_elements += sum(
+                            block.values.numel()
+                            for block in targets[target_key].blocks()
+                        )
+
+            self.scales[self.output_name_to_output_index[target_key]] = np.sqrt(
+                sum_of_squared_targets / total_num_elements
+            )
+
+    def restart(self, dataset_info: DatasetInfo) -> "Scaler":
+        # merge old and new dataset info
+        merged_info = self.dataset_info.union(dataset_info)
+
+        self.new_targets = {
+            key: value
+            for key, value in merged_info.targets.items()
+            if key not in self.dataset_info.targets
+        }
+
+        # register new outputs
+        for target_name, target in self.new_targets.items():
+            self._add_output(target_name, target)
+
+        self.dataset_info = merged_info
+
+        return self
+
+    def forward(
+        self,
+        outputs: Dict[str, TensorMap],
+    ) -> Dict[str, TensorMap]:
+        """
+        Scales all the targets in the outputs dictionary back to their
+        original scale.
+
+        :param outputs: A dictionary of target quantities and their values
+            to be scaled.
+
+        :raises ValueError: If an output does not have a corresponding
+            scale in the scaler model.
+        """
+
+        for output_name in outputs:
+            if output_name.startswith("mtt::aux::") or output_name == "features":
+                continue
```

Comment:
Should this be removed? Like for ZBL/Composition, the scaling should be explicitly picked by the model for each output

Reply:
Ok, it's gone